src/mito2/src/worker/handle_manifest.rs

```diff
@@ -294,6 +302,9 @@ impl<S> RegionWorkerLoop<S> {
 
         // Sets the region as writable.
         region.switch_state_to_writable(RegionLeaderState::Altering);
+        // Handles the stalled requests.
+        self.handle_region_stalled_requests(&change_result.region_id)
+            .await;
```

Comment:
Handle stalled requests after sending result to the sender so we won't block the waiter for too long.